decaf377-fmd/src/detection.rs

```diff
@@ -78,29 +78,29 @@ impl DetectionKey {
     /// This function executes in constant time with respect to the detection
     /// key material, but short-circuits to return early on a false detection.
     #[allow(non_snake_case)]
-    pub fn examine(&self, clue: &Clue) -> bool {
+    pub fn examine(&self, clue: &Clue) -> Result<(), Error> {
```

Comment:
This method should return a `bool` rather than a `Result`, since (unlike, e.g., a signature verification), getting a `false` result doesn't indicate an error, but just that the clue was not intended for this detection key.